File: src/main/java/ru/ioleksiv/telegram/bot/core/model/objects/inline/queryresult/TitledInlineResult.java
package ru.ioleksiv.telegram.bot.core.model.objects.inline.queryresult;

import com.fasterxml.jackson.annotation.JsonProperty;

public abstract class TitledInlineResult extends FormattedInlineResult {
    public final String getTitle() {
        return title;
    }

    public final void setTitle(String title) {
        this.title = title;
    }

    /**
     * title	String	Title for the result
     */
    @JsonProperty("title")
    private String title = null;

    TitledInlineResult(String type) {
        super(type);
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/objects/inline/queryresult/TitledInlineResult.kt
package io.github.oleksivio.tl.kbot.server.api.objects.inline.queryresult

import com.fasterxml.jackson.annotation.JsonProperty

abstract class TitledInlineResult : FormattedInlineResult() {
    /**
     * title String Title for the result
     */
    @JsonProperty("title")
    var title: String? = null
}
